source/distributed/tria.cc

```diff
@@ -3230,14 +3230,18 @@ namespace parallel
       Assert(cell_attached_data.pack_callbacks.size()==cell_attached_data.n_attached_datas,
              ExcMessage("register_data_attach(), not all data has been unpacked last time?"));
 
-      const unsigned int current_buffer_positition = cell_attached_data.cumulative_fixed_data_size+sizeof(CellStatus);
-      ++cell_attached_data.n_attached_datas;
+      // add new pair with offset and pack_callback
+      cell_attached_data.pack_callbacks.push_back(
+        std::make_pair(cell_attached_data.cumulative_fixed_data_size + sizeof(CellStatus),
+                       pack_callback)
+      );
+
+      // increase counters
       cell_attached_data.cumulative_fixed_data_size += size;
-      cell_attached_data.pack_callbacks[current_buffer_positition] = pack_callback;
 
-      // use the offset as the handle that callers receive and later hand back
-      // to notify_ready_to_unpack()
-      return current_buffer_positition;
+      // return the position of the callback in the register vector as a handle,
+      // then increment the value to get the correct amount of attached callbacks
+      return cell_attached_data.n_attached_datas++;
```

Comment:
For readability I would rather increment it above (where it was before)  and then return ‘... -1’